File: LeetCode/Easy/0219-contains-duplicate-ii/0219-contains-duplicate-ii.go
func containsNearbyDuplicate(nums []int, k int) bool {
    n := len(nums)
    d := min(n-1, k)

    m := make(map[int]int)

    for i := 0; i <= d; i++ {
        val, exists := m[nums[i]]
        if (exists && val == 1) {
            return true
        } else {
            m[nums[i]] = val + 1
        }
    }

    for i := 1; i + d < n; i++ {
        m[nums[i-1]] -= 1
        val, exists := m[nums[i+d]]
        if (exists && val == 1) {
            return true
        } else {
            m[nums[i+d]] = val + 1
        }
    }
    return false
}

File: LeetCode/Easy/0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int loc = map.getOrDefault(nums[i], -1);
            if (loc == -1) {
                map.put(nums[i], i);
                continue;
            }

            if (i - loc <= k) {
                return true;
            }
            map.put(nums[i], i);
        }
        return false;
    }
}